src/ess/reduce/nexus/generic_workflow.py

```diff
@@ -17,15 +17,15 @@
 from . import generic_types as gt
 from . import workflow
 from .generic_types import MonitorType, PulseSelection, RunType
-from .types import (
-    DetectorBankSizes,
-    GravityVector,
-    NeXusDetectorName,
-)
+from .types import DetectorBankSizes, GravityVector, NeXusDetectorName, PreopenNeXusFile
 
 
-def file_path_to_file_spec(filename: gt.Filename[RunType]) -> gt.NeXusFileSpec[RunType]:
-    return gt.NeXusFileSpec[RunType](filename)
+def file_path_to_file_spec(
+    filename: gt.Filename[RunType], preopen: PreopenNeXusFile
+) -> gt.NeXusFileSpec[RunType]:
+    return gt.NeXusFileSpec[RunType](
+        snx.File(filename, definitions=workflow.definitions) if preopen else filename
+    )
```

Comment:
I don't like this. Normally, we would open the file and pass the handle to a loader. I would expect the same from a workflow. That is, I would want to do this:
```python
handle = snx.File(...)
params[gt.Filename[SampleRun]] = handle
```

I thought the duck-typing approach would allow this?

Reply:
Well, that would not pass mypy, since `Filename` cannot be a file handle?

Reply:
Isn't this the whole point of not checking the types of parameters any more?

You are right that, in principle, it shouldn't pass. But it does pass.

Reply:
I am not sure I understand what you are saying. There appear to be 3 different things you mention: Having a provider open a file, mypy, and not checking types of parameters in Sciline.

They all seem kind of distinct. We should have a chat so you can explain!

I would like to control the behavior via a parameter, since having to change a ton of user code in a very complicated manner does not seem feasible or remotely user friendly.